--- a/config/waagent.conf
+++ b/config/waagent.conf
@@ -14,30 +14,34 @@
 Role.TopologyConsumer=None
 
 # Enable instance creation
-Provisioning.Enabled=y
+Provisioning.Enabled=n
 
 # Password authentication for root account will be unavailable.
-Provisioning.DeleteRootPassword=y
+Provisioning.DeleteRootPassword=n
 
 # Generate fresh host key pair.
-Provisioning.RegenerateSshHostKeyPair=y
+Provisioning.RegenerateSshHostKeyPair=n
 
 # Supported values are "rsa", "dsa" and "ecdsa".
 Provisioning.SshHostKeyPairType=rsa
 
 # Monitor host name changes and publish changes via DHCP requests.
-Provisioning.MonitorHostName=y
+Provisioning.MonitorHostName=n
 
 # Format if unformatted. If 'n', resource disk will not be mounted.
-ResourceDisk.Format=y
+ResourceDisk.Format=n
 
 # File system on the resource disk
 # Typically ext3 or ext4. FreeBSD images should use 'ufs2' here.
 ResourceDisk.Filesystem=ext4
 
 # Mount point for the resource disk
-ResourceDisk.MountPoint=/mnt/resource
+ResourceDisk.MountPoint=/mnt
 
+# NOTE: Ubuntu uses Cloud-init for disk-provisioning. This will
+# unless you disable Cloud-init disk provisioning. Please see
+# /usr/share/doc/walinuxagent/99-cloud-init-disable-diskprovisioning.conf
+#
 # Create and use swapfile on resource disk.
 ResourceDisk.EnableSwap=n
 
