Index: walinuxagent-2.2.2/config/ubuntu/waagent.conf
===================================================================
--- walinuxagent-2.2.2.orig/config/ubuntu/waagent.conf
+++ walinuxagent-2.2.2/config/ubuntu/waagent.conf
@@ -74,7 +74,7 @@ OS.OpensslPath=None
 # OS.EnableRDMA=y
 
 # Enable or disable goal state processing auto-update, default is enabled
-# AutoUpdate.Enabled=y
+AutoUpdate.Enabled=n
 
 # Determine the update family, this should not be changed
 # AutoUpdate.GAFamily=Prod
