--- a/config/waagent.conf
+++ b/config/waagent.conf
@@ -3,16 +3,16 @@
 #
 
 # Enable instance creation
-Provisioning.Enabled=y
+Provisioning.Enabled=n
 
 # Rely on cloud-init to provision
 Provisioning.UseCloudInit=n
 
 # Password authentication for root account will be unavailable.
-Provisioning.DeleteRootPassword=y
+Provisioning.DeleteRootPassword=n
 
 # Generate fresh host key pair.
-Provisioning.RegenerateSshHostKeyPair=y
+Provisioning.RegenerateSshHostKeyPair=n
 
 # Supported values are "rsa", "dsa" and "ecdsa".
 Provisioning.SshHostKeyPairType=rsa
@@ -36,14 +36,14 @@
 Provisioning.AllowResetSysUser=n
 
 # Format if unformatted. If 'n', resource disk will not be mounted.
-ResourceDisk.Format=y
+ResourceDisk.Format=n
 
 # File system on the resource disk
 # Typically ext3 or ext4. FreeBSD images should use 'ufs2' here.
 ResourceDisk.Filesystem=ext4
 
 # Mount point for the resource disk
-ResourceDisk.MountPoint=/mnt/resource
+ResourceDisk.MountPoint=/mnt
 
 # Create and use swapfile on resource disk.
 ResourceDisk.EnableSwap=n
