--- a/distro/systemd/waagent.service
+++ /dev/null
@@ -1,13 +0,0 @@
-[Unit]
-Description=Windows Azure Linux Agent
-After=network.target
-After=sshd.service
-ConditionFileIsExecutable=/usr/sbin/waagent
-ConditionPathExists=/etc/waagent.conf
-
-[Service]
-Type=simple
-ExecStart=/usr/sbin/waagent -daemon
-
-[Install]
-WantedBy=multi-user.target
--- /dev/null
+++ b/distro/systemd/walinuxagent.service
@@ -0,0 +1,13 @@
+[Unit]
+Description=Windows Azure Linux Agent
+After=network.target
+Wants=sshd.service sshd-keygen.service
+ConditionFileIsExecutable=/usr/sbin/waagent
+ConditionPathExists=/etc/waagent.conf
+
+[Service]
+Type=simple
+ExecStart=/usr/sbin/waagent -daemon
+
+[Install]
+WantedBy=multi-user.target
