Description: Use the proper permissions on /etc/shadow (LP: #1188820).
Bug: https://bugs.launchpad.net/ubuntu/+source/walinuxagent/+bug/1188820
Author: Ben Howard
Last-Update: 2013-07-23
--- a/waagent
+++ b/waagent
@@ -2423,7 +2423,7 @@ def DeleteRootPassword():
     ReplaceFileContentsAtomic(filepath, "root:*LOCK*:14600::::::\n" + "\n".join(filter(lambda a: not
         a.startswith("root:"),
         GetFileContents(filepath).split('\n'))))
-    os.chmod(filepath, 0000)
+    os.chmod(filepath, 0640)
     if IsRedHat():
         Run("chcon system_u:object_r:shadow_t:s0 " + filepath)
     Log("Root password deleted.")
